File: 2018/puzzle1/js/solution.js
const fs = require('fs')

fs.readFile('../input.txt', 'utf8', (err, data) => {
    let numbers = data.split('\n').map(numStr => Number(numStr))
    let sumTotal = numbers.reduce((total, num) => total + num)
    console.log(
        sumTotal >= 0 ?
        `The total change in frequency is +${sumTotal}` :
        `The total change in frequency is ${sumTotal}`
    )
})

File: 2018/python/puzzle1/solution.py
"""Advent of Code, Puzzle 1 solution."""
if __name__ == "__main__":
    with open('input.txt') as data:
        total = sum([int(number) for number in data.readlines()])
        print(f"The total change in frequency is +{total}" if total > 0 else f"The total change in frequency is {total}")
